core/server/master/src/main/java/alluxio/master/backup/BackupLeaderRole.java

```diff
@@ -0,0 +1,293 @@
+    workerConnection.handler(BackupHeartbeatMessage.class,
+        (Consumer<BackupHeartbeatMessage>) request -> handleHeartbeatMessage(request));
+    // Register connection error listener.
+    workerConnection.onException((error) -> {
+      LOG.warn(String.format("Worker connection failed for %s.", workerConnection), error);
+    });
+    // Register connection close listener.
+    workerConnection.onClose((conn) -> {
+      LOG.info("Worker connection closed for {}.", workerConnection);
+      // Remove the connection when completed
+      mFollowerConnections.remove(conn);
+      // Fail current backup if it was driven by the closed connection.
+      if (mBackupControl != null && mBackupControl.getFirst() == conn) {
+        LOG.warn("Abandoning current backup as worker is lost.");
+        mBackupTracker.updateError(new BackupAbortedException("Backup worker lost."));
+        mBackupControl = null;
+      }
+    });
+  }
+
+  /**
+   * Schedule backup on this master.
+   */
+  private void scheduleLocalBackup(BackupPRequest request) {
+    Future<?> backupFuture = mExecutorService.submit(() -> {
+      try (LockResource lr = new LockResource(mStatePauseLock)) {
+        try {
+          mBackupTracker.updateState(BackupState.RUNNING);
+          AlluxioURI backupUri = takeBackup(request, mBackupTracker.getEntryCounter());
+          mBackupTracker.updateBackupUri(backupUri);
+          mBackupTracker.updateState(BackupState.COMPLETED);
+        } catch (IOException e) {
+          mBackupTracker.updateError(new BackupException("Local backup failed.", e));
+        }
+      }
+    });
+    // Set backup control for self execution.
+    mBackupControl = new Pair(null, backupFuture);
+  }
+
+  /**
+   * Delegates a backup to a worker.
+   *
+   * After successful delegation, follower will send regular heartbeats to leader for updating the
+   * backup status.
+   *
+   * @return {@code true} if delegation successful
+   */
+  private boolean scheduleRemoteBackup(BackupPRequest request) {
+    // Try to delegate backup to a follower.
+    for (Connection connection : mFollowerConnections) {
+      try {
+        // Suspend journals on current follower.
+        sendMessageBlocking(connection, new BackupSuspendMessage());
+        // Get consistent journal sequences.
+        Map<String, Long> journalSequences;
+        try (LockResource stateLock = new LockResource(mStatePauseLock)) {
+          journalSequences = mJournalSystem.getCurrentSequences();
+        }
+        // Send backup request along with consistent journal sequences.
+        sendMessageBlocking(connection, new BackupRequestMessage(request, journalSequences));
+        // Delegation successful.
+        mBackupControl = new Pair<>(connection, null);
+        // Start abandon timer.
+        adjustAbandonTimeout(false);
+        return true;
+      } catch (Exception e) {
+        LOG.warn(String.format("Failed to delegate backup to worker: %s", connection), e);
```

Comment:
The same. in Alluxio logs, user may think it's Alluxio worker instead of backup worker.